File: Solution/1137.第N个泰波那契数/1137.第N个泰波那契数.py
'''
Descroption: LeetCode 1137. 第 N 个泰波那契数
Author: EmoryHuang
Date: 2021-07-17 15:11:33
解题思路:
通过题目给的公式计算即可
'''


class Solution:
    def tribonacci(self, n: int) -> int:
        a, b, c = 0, 1, 1
        for _ in range(n):
            a, b, c = b, c, a + b + c
        return a

File: Solution/1137.第N个泰波那契数/1137.第N个泰波那契数.java
/*
 * @Descroption: LeetCode 1137. 第 N 个泰波那契数
 * @Author: EmoryHuang
 * @Date: 2023-07-21 10:07:50
 * @Method:
 * 动态规划
 */

import java.util.*;

class Solution {
    public int tribonacci(int n) {
        if (n == 0) return 0;
        if (n <= 2) return 1;
        int a = 0, b = 1, c = 1;
        for (int i = 3; i <= n; i++) {
            int temp = a;
            a = b;
            b = c;
            c = temp + a + b;
        }
        return c;
    }
}